mongomock/collection.py

```diff
@@ -1553,18 +1575,10 @@ def _extend_collection(out_collection, field, expression):
                                     key_getter = functools.partial(_parse_expression, key)
                                     values = [key_getter(doc) for doc in group_list]
 
-                                    if operator == "$sum":
-                                        val_it = (val or 0 for val in values)
-                                        doc_dict[field] = sum(val_it)
-                                    elif operator == "$avg":
-                                        values = [val or 0 for val in values]
-                                        doc_dict[field] = sum(values) / max(len(values), 1)
-                                    elif operator == "$min":
-                                        val_it = (val or MAXSIZE for val in values)
-                                        doc_dict[field] = min(val_it)
-                                    elif operator == "$max":
-                                        val_it = (val or -MAXSIZE for val in values)
-                                        doc_dict[field] = max(val_it)
+                                    if operator in aggregate_operators:
+                                        doc_dict[field] = _handle_aggregate_operator(operator,
+                                                                                     values,
+                                                                                     doc_dict)
                                     elif operator == "$first":
```

Comment:
I suggest you use the `group_operators` few line above (l.1565 in your new code) and that you have the final else to that chain here handling group operators that are not implemented yet.